src/selectors/operators/compares/valueCompare.js

```diff
@@ -1,24 +1,113 @@
 import castToType from '../../dataTypes/castToType';
 import isInstanceOfType from '../../dataTypes/isInstanceOfType';
+import DateTime from '../../dataTypes/valueTypes/DateTime';
+import YearMonthDuration from '../../dataTypes/valueTypes/YearMonthDuration';
+import DayTimeDuration from '../../dataTypes/valueTypes/DayTimeDuration';
 
 /**
  * @param  {string}    operator
  * @param  {../../dataTypes/Sequence}  firstSequence
  * @param  {../../dataTypes/Sequence}  secondSequence
  */
-export default function valueCompare (operator, firstSequence, secondSequence) {
-    // https://www.w3.org/TR/xpath-3/#doc-xpath31-ValueComp
+export default function valueCompare (operator, firstSequence, secondSequence, dynamicContext) {
+	// https://www.w3.org/TR/xpath-3/#doc-xpath31-ValueComp
 	if (!firstSequence.isSingleton() || !secondSequence.isSingleton()) {
-        throw new Error('XPTY0004: Sequences to compare are not singleton');
-    }
+		throw new Error('XPTY0004: Sequences to compare are not singleton');
+	}
+
+	let firstValue = firstSequence.first();
+	let secondValue = secondSequence.first();
+
+	const bothInstanceOfType = type => isInstanceOfType(firstValue, type) && isInstanceOfType(secondValue, type);
+
+	if (bothInstanceOfType('xs:dateTime') ||
```

Comment:
In the future, it is better to structure your conditionals to make the most common case the first. ie: first numerics, then untypedAtomic, etc, etc, etc. This will have to be re-written before rebasing.